### DISCUSSION TITLE
New to PiSCSI and can't use WEBUI

### DISCUSSION THREAD

**Discussion by jzmacdaddy** [akuker/RASCSI#1258] [General]
I used the latest image on my Pi Zero W. It boots up and appears to work, but I only get the waiting screen on the Web UI. When I run  "journalctl -f" the only error I see is:

cannot import name 'url_quote' from 'werkzeug.urls'

When I tried the pre-built v22.10.01 version, the web interface worked fine.

**Comment by rdmark** [comment]
I apologize for the trouble! This is a known issue that has been patched with https://github.com/PiSCSI/piscsi/issues/1233

We will work towards a new release shortly.

If you're handy on the Linux command line, you can easily fix the problem in your local environment. Please let us know if you want to attempt it and I can walk you through the steps.

**Comment by mooneyjess** [comment]
I am having this same issue, it stays stuck on the "PiSCSI Web Interface is Starting...." page.

I get the error when using the journalctl -f command;
**ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/home/pi/piscsi/python/web/venv/lib/python3.9/site-packages/werkzeug/urls.py)**

I added werkzeug==2.3.7 to the requirements.txt in the vi editor and it looks like this (located in _/piscsi/python/web_);

bjoern==3.2.2
Flask==2.2.2
Jinja2==3.1.2
protobuf==3.20.2
requests==2.31.0
simplepam==0.1.5
flask_babel==2.0.0
ua-parser==0.16.1
vcgencmd==0.1.1
werkzeug==2.3.7

Is there a step that I am missing? I have been trying to figure this out for a few days.

Here is some extra info;

piscsi.service - PiSCSI service
     Loaded: loaded (/etc/systemd/system/piscsi.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-11-10 09:58:54 EST; 11min ago
   Main PID: 518 (piscsi)
      Tasks: 2 (limit: 4915)
        CPU: 87ms
     CGroup: /system.slice/piscsi.service
             └─518 /usr/local/bin/piscsi -r 7 -F /home/pi/images

Nov 10 09:58:55 raspberrypi PISCSI[518]: Version 23.04.01  (Nov  9 2023 22:54:47)
Nov 10 09:58:55 raspberrypi PISCSI[518]: Powered by XM6 TypeG Technology / Copyright (C) 2016-2020 GIMONS
Nov 10 09:58:55 raspberrypi PISCSI[518]: Copyright (C) 2020-2023 Contributors to the PiSCSI project
Nov 10 09:58:55 raspberrypi PISCSI[518]: Connection type: FULLSPEC
Nov 10 09:58:55 raspberrypi PISCSI[518]: [2023-11-10 09:58:55.214] [info] Set log level to 'info'
Nov 10 09:58:55 raspberrypi PISCSI[518]: [2023-11-10 09:58:55.215] [info] Detected device Raspberry Pi 4
Nov 10 09:58:55 raspberrypi PISCSI[518]: [2023-11-10 09:58:55.288] [info] Reserved ID(s) set to 7
Nov 10 09:58:55 raspberrypi PISCSI[518]: [2023-11-10 09:58:55.301] [info] Default image folder set to '/home/>
Nov 10 09:58:55 raspberrypi PISCSI[518]: [2023-11-10 09:58:55.302] [info] No devices currently attached.
Nov 10 09:58:55 raspberrypi PISCSI[518]: No devices currently attached.

**Comment by rdmark** [comment]
Did you read the steps on the release notes page on how to recover from this error?

https://github.com/PiSCSI/piscsi/releases/tag/v23.04.01

**Reply by mooneyjess** [reply]
I could not find those steps before; however, with the November 2023 update it is now working fine. It says that I am on PiSCSI software version: 23.11.1. Looks like before I was running 23.04.01 (the April version).

**Reply by rdmark** [reply]
That's good to hear! Yes the Nov 2023 is a pre-release version that I tagged yesterday. It should be good to use, but I'm having problems building the pre-built RPi images so it isn't "officially" released yet.
